packages/cli/src/models/compiler/Compiler.ts

```diff
@@ -10,7 +10,7 @@ const Compiler = {
     if (!FileSystem.exists(truffleBin)) truffleBin = 'truffle'; // Attempt to load global truffle if local was not found
 
     return new Promise((resolve, reject) => {
-      exec(`${truffleBin} compile --all`, (error, stdout, stderr) => {
+      execFile(truffleBin, ['compile', '--all'], (error: ExecException, stdout, stderr) => {
```

Comment:
WDYT if we simplify `Compiler#call` to something prettier like:

```ts
import { promisify } from 'util';
const execFileAsync = promisify(execFile);

const Compiler = {
  async call(): Promise<void | ExecException> {
    let truffleBin = `${process.cwd()}/node_modules/.bin/truffle`;
    log.info('Compiling contracts with Truffle...');
    if (!FileSystem.exists(truffleBin)) truffleBin = 'truffle'; // Attempt to load global truffle if local was not found
    try {
      const { stdout } = await execFileAsync(truffleBin, ['compile', '--all']);
      log.warn(stdout); //or log.info?
    } catch(error) {
      throw Error(error.stdout);
    }
  }
};
```
I believe that in both cases (whether or not there is a compilation error) `stderr` will be empty, and the output will be in `stdout`

Not a blocker though.